### PR TITLE
Add guidance for Next.js

### PR DESCRIPTION
**Pull request by haydn**
This adds a section to `readme.md` with some advice for people using Next.js. IMO, having a dedicated config would be a burden to maintain without adding much value (mainly because Next.js rewrites the `tsconfig.json`). Having some guidance should be enough to point people into the pit of success.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -93,9 +93,36 @@ Mostly relevant for when you're transpiling with `tsc`. If you want to change th
...
+- `plugins` (adds the `"next"` plugin if not present)
+- `strictNullChecks`
+
+It will also add `"next-env.d.ts"` and `".next/types/**/*.ts"` to the `include` field so the Next.js types are available. The final config file will look like this:
+
+```json
+{
+  "extends": "@total-typescript/tsconfig/bundler/dom/app",
+  "compilerOptions": {
+    "incremental": true,
```
Comment (GermanJablo) [inline readme.md:116]: related: https://github.com/total-typescript/tsconfig/issues/15
Is there any case where this option should NOT be default?

**Comment by mattpocock** [comment]
Thanks for the pull request! However, I think it makes more sense for frameworks themselves to own the documentation for how they work best with TypeScript.

File: readme.md (hunk 1 above)
Comment (haydn) [reply readme.md:116]: I believe even if you turn it off, Next.js will turn it back on. You don't really get a choice in the matter.

**Comment by haydn** [comment]
@mattpocock No worries, that makes sense. 👍

**Pull request closed without merging**